### ISSUE TITLE
Does the strlen function need to be recalculated every round of the for loop?

### ISSUE THREAD

**Issue by Mserhatarslan** [ritwik12/Virtual-Assistant#138]
strlen(example) function is called in each iteration of the loop to determine the length of the string example. However, calculating the length of the string repeatedly in each iteration can be inefficient, especially if the length remains constant throughout the loop.

[Virtual-Assistant/analysis.c](https://github.com/ritwik12/Virtual-Assistant/blob/2712a8917b4480e96350af33e58fd1e59c9192f2/analysis.c#L15)

for (int iter_char = 0; iter_char < strlen(example); iter_char++) {

```
int len = strlen(example);

for (int iter_char = 0; iter_char < len; iter_char++) {
    example[iter_char] = tolower(example[iter_char]);
    if (example[iter_char] == ' ') {
        if (example[iter_char + 1] != ' ') {
            split[word][character] = '\0';
            character = 0;
            word++;
        }
        continue;
    } else {
        split[word][character++] = example[iter_char];
    }
}
```

By calculating the length of the string outside the loop and storing it in the variable len, you avoid the overhead of recomputing the length in each iteration. This can lead to a significant improvement in performance, especially for long strings.

**Comment by mdahamshi** [comment]
I agree with that, you want to make sure example length doesn't change until the loop is done...

**Comment by ritwik12** [comment]
Agreed, can be worked upon. thanks for the suggestion. Waiting for someone to pick this up
